pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/__init__.py

```diff
@@ -112,7 +112,7 @@ def parser_warn(msg):
 
     # TODO: use deprecated=True in Python >=3.13
     if args.no_ssh_tty:
-        parser_warn("--no-ssh-tty deprecated, SSH's pseudo-TTY is never used anymore")
+        parser_warn("--no-ssh-tty deprecated, SSH's TTY is never used anymore")
```

Comment:
Isn't it technically still used for the sudo prompt? Maybe reword this:

`--no-ssh-tty deprecated, a TTY is only allocated if it's explicitly needed`

Reply:
No, SSH's TTY (the `-t` flag) is never used anymore, this is why we have `--ask-sudo-password` (that asks the password using Python's `getpass` instead and pass the password via stdin using `--stdin` flag in `sudo`).

The reason for it is because SSH's TTY mess up with lots of the expectations you have with SSH, e.g.: it combines stdout and stderr in one stream (stdout), so if you expect a command to print something to stderr for example it will be wrong.

Of course this can be workaround, but it is better to simple not do it. The bonus is that using something like `--ask-sudo-password` means we avoid the previous issue where the `sudo` password were asked twice because each SSH connection created a new TTY and by default the SSH cache is associated to each TTY (it can be changed, but it is rarely done so). Also this is the way other tools do it, for example Ansible has a `--ask-become-pass` that is similar behavior of `--ask-sudo-password`.

Reply:
Oh, I see. I misread the code earlier.